sysid-application/src/main/native/cpp/view/Analyzer.cpp

```diff
@@ -751,28 +751,17 @@ void Analyzer::DisplayFeedbackGains() {
   // Come back to the starting y pos.
   ImGui::SetCursorPosY(beginY);
 
-  auto ShowLQRParam = [this](const char* text, double* data, float min,
-                             float max, bool power = true) {
-    float val = *data;
-    ImGui::SetNextItemWidth(ImGui::GetFontSize() * 5);
-    ImGui::SetCursorPosX(ImGui::GetFontSize() * 9);
-
-    if (ImGui::SliderFloat(text, &val, min, max, "%.1f",
-                           ImGuiSliderFlags_None |
-                               (power ? ImGuiSliderFlags_Logarithmic : 0))) {
-      *data = static_cast<double>(val);
-      UpdateFeedbackGains();
-    }
-  };
-
   std::string_view abbreviation = GetAbbreviation(m_manager->GetUnit());
 
   if (m_selectedLoopType == 0) {
-    ShowLQRParam(fmt::format("Max Position Error ({})", abbreviation).c_str(),
-                 &m_settings.lqr.qp, 0.05f, 40.0f);
+    ImGui::SetCursorPosX(ImGui::GetFontSize() * 9);
+    DisplayGain(fmt::format("Max Position Error ({})", abbreviation).c_str(),
+                &m_settings.lqr.qp, false);
   }
 
-  ShowLQRParam(fmt::format("Max Velocity Error ({}/s)", abbreviation).c_str(),
-               &m_settings.lqr.qv, 0.05f, 40.0f);
-  ShowLQRParam("Max Control Effort (V)", &m_settings.lqr.r, 0.1f, 12.0f, false);
+  ImGui::SetCursorPosX(ImGui::GetFontSize() * 9);
+  DisplayGain(fmt::format("Max Velocity Error ({}/s)", abbreviation).c_str(),
+              &m_settings.lqr.qp, false);
+  ImGui::SetCursorPosX(ImGui::GetFontSize() * 9);
+  DisplayGain("Max Control Effort (V)", &m_settings.lqr.r, false);
```

Comment:
Could we add a maximum voltage cap for 12 V? I think just a simple if statement setting m_settings.lqr.r to 12 if its greater than it should work.

Reply:
I don't think there's really a reason to do this; it's fun to play with the tool to see what happens in the limiting cases, and you can always rescale both terms without changing the LQR results

Reply:
Alright that's fair